bycycle/utils.py

```diff
@@ -0,0 +1,90 @@
+"""Plotting utility functions."""
+
+import numpy as np
+
+###################################################################################################
+###################################################################################################
+
+
+def limit_df(df, fs, xlim):
+    """Restrict dataframe to be within time limits.
+
+    Parameters
+    ----------
+    df : pandas DataFrame
+        Dataframe output of :func:`~.compute_features`.
+    fs : float
+        Sampling rate, in Hz.
+    xlim : tuple of (float, float), optional, default: None
+        Start and stop times.
+
+    Returns
+    -------
+    df : pandas DataFrame
+        A limited dataframe of cycle features.
+    """
+
+    center_e, side_e = get_extrema(df)
+
+    # Limit dataframe to xlim
+    df = df[(df['sample_next_' + side_e].values >= xlim[0]*fs) &
+            (df['sample_last_' + side_e].values < xlim[1]*fs)]
+
+    # Shift sample indices to start at 0
+    df['sample_last_' + side_e] = df['sample_last_' + side_e] - int(fs * xlim[0])
+    df['sample_next_' + side_e] = df['sample_next_' + side_e] - int(fs * xlim[0])
+    df['sample_' + center_e] = df['sample_' + center_e] - int(fs * xlim[0])
+    df['sample_zerox_rise'] = df['sample_zerox_rise'] - int(fs * xlim[0])
+    df['sample_zerox_decay'] = df['sample_zerox_decay'] - int(fs * xlim[0])
+
+    return df
+
+
+def limit_sig_times(sig, times, xlim):
```

Comment:
A bit pedantic, but this doesn't just limit the _times_, but rather the signal too. 
Maybe call is `limit_signal`?

Reply:
We can call rename it to `limit_signal`. I went with the `sig_times` because it limits and returns both.

Reply:
Ohhh, ooops. 

I sorta mis-read it as "limit the times vector of the signal" rather than "limit sig & times". I see what the original name was now. 

I think I still vote for `limit_signal`, though, as that seems to me to be the goal / purpose (`times` is sorta just meta-data).

Reply:
I'll change it to `limit_signal`. I think the original name could easily be misinterpreted.